File: prettier-plugin-csharpier/src/Printer/Types/QueryExpression.ts
import { PrintMethod } from "../PrintMethod";
import { SyntaxTreeNode } from "../SyntaxTreeNode";
import { concat, group, hardline, indent, join, softline, line, doubleHardline } from "../Builders";
import { FromClauseNode, printFromClause } from "./FromClause";
import { printQueryBody, QueryBodyNode } from "./QueryBody";

export interface QueryExpressionNode extends SyntaxTreeNode<"QueryExpression"> {
    fromClause: FromClauseNode;
    body: QueryBodyNode;
}

export const printQueryExpression: PrintMethod<QueryExpressionNode> = (path, options, print) => {
    return concat([
        path.call(o => printFromClause(o, options, print), "fromClause"),
        indent(concat([line, path.call(o => printQueryBody(o, options, print), "body")]))
    ]);
};

File: Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/QueryExpression.cs
namespace CSharpier.SyntaxPrinter.SyntaxNodePrinters;

internal static class QueryExpression
{
    public static Doc Print(QueryExpressionSyntax node, PrintingContext context)
    {
        return Doc.Concat(
            FromClause.Print(node.FromClause, context),
            Doc.Line,
            QueryBody.Print(node.Body, context)
        );
    }
}
